packages/flutter_tools/test/android/gradle_test.dart

```diff
@@ -331,16 +330,6 @@ flutter:
   });
 }
 
-void writeSchemaFile(FileSystem filesystem, String schemaData) {
-  final String schemaPath = buildSchemaPath(filesystem);
-  final File schemaFile = filesystem.file(schemaPath);
-
-  final String schemaDir = buildSchemaDir(filesystem);
-
-  filesystem.directory(schemaDir).createSync(recursive: true);
-  filesystem.file(schemaFile).writeAsStringSync(schemaData);
```

Comment:
Weird - in the old code (which I moved to be shared) we're passing a `File` into `filesystem.file()`. The `file` method takes `dynamic` so this didn't produce any errors. I noticed because I found another(!) copy of this function and it was slightly different again.

Reply:
Apparently by design:

> path can be either a `String`, a `Uri`, or a `FileSystemEntity`.

I thought maybe it was re-using the instance as a performance thing, however it reads the `.path` of it and constructors another one, so I guess not.